src/renderer/components/recording-page/recording-visualization.tsx

```diff
@@ -2,134 +2,162 @@ import log from 'electron-log';
 import React, { FC, useEffect, useRef } from 'react';
 import Col from 'react-bootstrap/Col';
 import Row from 'react-bootstrap/Row';
-import useInterval from 'react-use/lib/useInterval';
 import WaveSurfer from 'wavesurfer.js';
+import MicrophonePlugin from 'wavesurfer.js/dist/plugin/wavesurfer.microphone';
 import SpectrogramPlugin from 'wavesurfer.js/dist/plugin/wavesurfer.spectrogram';
 
 import { noOp } from '../../../common/env-and-consts';
-import { checkFileExistence, join, readFile } from '../../utils';
+import { acquireAudioInputStream, checkFileExistence, join, readFile } from '../../utils';
+import { FileMonitor } from '../../utils/file-monitor';
+import { useAudioInputOutputDevices } from '../settings-page/hooks';
 
 import { State } from './types';
 
 interface RecordingVisualizationProps {
   prevState?: State;
   state: State;
   filePath?: string;
+  basePath: string;
+  recItemName: string;
 }
 
 const GRAPH_WIDTH = '60%';
-const GRAPH_HEIGHT = 80;
-const EMPTY_IMAGE_URL = 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==';
 
-export const RecordingVisualization: FC<RecordingVisualizationProps> = ({ prevState, state, filePath }) => {
+export const RecordingVisualization: FC<RecordingVisualizationProps> = ({
+  prevState,
+  state,
+  filePath,
+  basePath,
+  recItemName,
+}) => {
   const waveSurferRef = useRef<WaveSurfer>();
   const waveformContainerRef = useRef<HTMLDivElement>(null);
-  const waveformImageRef = useRef<HTMLImageElement>(null);
   const spectrogramContainerRef = useRef<HTMLDivElement>(null);
-  const spectrogramImageRef = useRef<HTMLImageElement>(null);
+  const [, , audioInputDeviceId, , ,] = useAudioInputOutputDevices();
+  const fileMonitor = new FileMonitor(basePath);
```

Comment:
You probably don't want to create a new FileMonitor when the component is rerendered. Use `useInitializerRef`

Reply:
okay so I'm working on switching this to an useInitializerRef (which I've never used so I'm basing this off of an example from one of your other files) - would I declare the file monitor instance like I have it now and just move the subscription to the useInitializerRef?

Reply:
You need to pass it a factory function that takes 0 arguments:
```typescript
useInitializerRef(() => new FileMonitor(basePath))
```